driver-core/src/main/com/mongodb/connection/DefaultServer.java

```diff
@@ -177,6 +181,35 @@ public void onResult(final T result, final Throwable t) {
                 }
             }, LOGGER));
         }
+
+        @Override
+        public <T> T execute(final CommandProtocol<T> protocol, final InternalConnection connection,
+                             final SessionContext sessionContext) {
+            try {
+                protocol.sessionContext(new ClusterClockAdvancingSessionContext(sessionContext));
```

Comment:
Rubber hits road: The session context is wrapped in a private one that take the singleton ClusterClock into account.

Reply:
Ack.